src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java

```diff
@@ -129,19 +139,56 @@ public void run(ActionContext context) throws Exception {
 
     QueryJobConfiguration queryConfig = builder.build();
 
-    Job queryJob = bigQuery.create(JobInfo.newBuilder(queryConfig).setJobId(jobId).build());
+    // Exponential backoff with initial retry of 1 second and max retry of 32 seconds.
+    executeQueryWithExponentialBackoff(Duration.ofSeconds(1), Duration.ofSeconds(32), 2, 5,
+            bigQuery, queryConfig, context);
+  }
+
+  protected void executeQueryWithExponentialBackoff(Duration initialRetryDuration, Duration maxRetryDuration,
```

Comment:
Instead of this, write one method which returns Retry Policy without any input from previous method as we are not taking anything from user. and call executeQuery from run itself using that retry policy.

Reply:
Made a new method